cmd/mt-whisper-importer-reader/main.go

```diff
@@ -304,52 +303,42 @@ func getMetric(w *whisper.Whisper, file, name string) (archive.Metric, error) {
 		points[i] = p
 	}
 
+	_, selectedSchema := schemas.Match(res.MetricData.Name, int(w.Header.Archives[0].SecondsPerPoint))
 	conversion := newConversion(w.Header.Archives, points, method)
-	for retIdx, retention := range schem.Retentions {
+	for retIdx, retention := range selectedSchema.Retentions {
 		convertedPoints := conversion.getPoints(retIdx, uint32(retention.SecondsPerPoint), uint32(retention.NumberOfPoints))
 		for m, p := range convertedPoints {
 			if len(p) == 0 {
 				continue
 			}
-			mkey, err := schema.MKeyFromString(md.Id)
+
+			mkey, err := schema.MKeyFromString(res.MetricData.Id)
```

Comment:
res.MetricData.Id doesnt change.  We should get the mkey once before looping through the retentions and points.